### ISSUE TITLE
IAuthorizationFilter blocking Smidge route

### ISSUE THREAD

**Issue by paulodiogo** [Shazwazza/Smidge#116]
When I try to use a IAuthorizationFilter implementation, I had to skip when `context.RouteData?.Values["bundle"] != null` .

There is a solution for this? Maybe I did some misconfiguration?

**Comment by Shazwazza** [comment]
Can you please provide the basic steps to replicate?

**Comment by paulodiogo** [comment]
Create a IAuthorizationFilter and add it as Global:

```
public class GlobalAuthorizationFilter : IAuthorizationFilter
    {
        public void OnAuthorization(AuthorizationFilterContext context)
        {
            if (!context.HttpContext.User.Identity.IsAuthenticated)
            {
                context.Result = new UnauthorizedResult();
                return;
            }
			//CODE
        }
    }


//at Startup.ConfigureServices(IServiceCollection services)

  services.AddControllersWithViews(options =>
  {
      options.Filters.Add(typeof(GlobalAuthorizationFilter));
  });
```

**Comment by Shazwazza** [comment]
So you are adding a global auth filter to all controllers which will include Smidge controllers and therefore the Smidge controllers will not execute. 

A fix could be that we include the `[AllowAnonymous]` attribute on the smidge controller. I think this controller should always be publicly available since it's just used for returning css/js. 

You should be able to determine the current controller executing in your route values too and could just check for "Smidge" as the controller name and exclude your authz. 

Let me know what you think.

**Comment by Shazwazza** [comment]
I Just published a new 4.0 beta release with the [AllowAnonymous] change  if you want to check it out https://www.nuget.org/packages/Smidge.Nuglify/4.0.0-beta.357

**Comment by paulodiogo** [comment]
> So you are adding a global auth filter to all controllers which will include Smidge controllers and therefore the Smidge controllers will not execute.
> 
> A fix could be that we include the `[AllowAnonymous]` attribute on the smidge controller. I think this controller should always be publicly available since it's just used for returning css/js.
> 
> You should be able to determine the current controller executing in your route values too and could just check for "Smidge" as the controller name and exclude your authz.
> 
> Let me know what you think.

Tthis is exactly what I have done. Will check it out the beta version. Thank you!
